src/contracts/pods/EigenPodManager.sol

```diff
@@ -134,68 +184,142 @@ contract EigenPodManager is Initializable, OwnableUpgradeable, Pausable, IEigenP
+        onlyNotFrozen(queuedWithdrawal.delegatedAddress)
+        nonReentrant
+        onlyWhenNotPaused(PAUSED_WITHDRAWALS)
+    {
+        _completeWithdrawal(queuedWithdrawal, middlewareTimesIndex);
     }
+    
+    function slashShares(
+        address slashedPodOwner,
+        address slashedFundsRecipient,
+        uint256 shareAmount
+    )
+        external
+        onlyOwner
+        onlyNotFrozen(slashedPodOwner)
+        nonReentrant
+    {
+        require(shareAmount > 0, "EigenPodManager.slashShares: shares must be greater than zero");
+        require(podOwnerShares[slashedPodOwner] >= shareAmount, "EigenPodManager.slashShares: podOwnerShares[podOwner] must be greater than or equal to shares");
 
-    /**
-     * @notice This function is called to decrement withdrawableRestakedExecutionLayerGwei when a validator queues a withdrawal.
-     * @param amountWei is the amount of ETH in wei to decrement withdrawableRestakedExecutionLayerGwei by
-     */
-    function decrementWithdrawableRestakedExecutionLayerGwei(address podOwner, uint256 amountWei) 
-        external onlyStrategyManager onlyWhenNotPaused(PAUSED_WITHDRAW_RESTAKED_ETH) 
+        _removeShares(slashedPodOwner, shareAmount);
+        _withdrawRestakedBeaconChainETH(slashedPodOwner, slashedFundsRecipient, shareAmount);
+
+        uint256[] memory shareAmounts = new uint256[](1);
+        shareAmounts[0] = shareAmount;
+        IStrategy[] public strategies = new IStrategy[](1);
+        strategies[0] = beaconChainETHStrategy;
+        delegationManager.decreaseDelegatedShares(slashedPodOwner, strategies, shareAmounts);
+    }
+
+    function slashQueuedWithdrawal(
+        address slashedFundsRecipient,
+        BeaconChainQueuedWithdrawal memory queuedWithdrawal
+    )
+        external 
+        onlyOwner 
+        onlyFrozen(queuedWithdrawal.delegatedAddress) 
+        nonReentrant 
     {
-        ownerToPod[podOwner].decrementWithdrawableRestakedExecutionLayerGwei(amountWei);
+        // find the withdrawalRoot
+        bytes32 withdrawalRoot = _calculateWithdrawalRoot(queuedWithdrawal);
+
+         // verify that the queued withdrawal is pending
+        require(
+            withdrawalRootPending[withdrawalRoot],
+            "EigenPodManager.slashQueuedWithdrawal: withdrawal is not pending"
+        );
+
+        // reset the storage slot in mapping of queued withdrawals
+        withdrawalRootPending[withdrawalRoot] = false;
+
+        _withdrawRestakedBeaconChainETH(queuedWithdrawal.podOwner, slashedFundsRecipient, queuedWithdrawal.shares);
     }
 
     /**
-     * @notice This function is called to increment withdrawableRestakedExecutionLayerGwei when a validator's withdrawal is completed.
-     * @param amountWei is the amount of ETH in wei to increment withdrawableRestakedExecutionLayerGwei by
+     * @notice forces a withdrawal of the podOwner's beaconChainETHStrategy shares
+     * @param podOwner is the pod owner whose shares are to be removed
```

Comment:
probably want a dev tag here which notes that only the DelegationManager can call this function